packages/interactivity-router/src/index.js

```diff
@@ -102,12 +135,47 @@ window.addEventListener( 'popstate', async () => {
 	}
 } );
 
-// Cache the initial page using the intially parsed vDOM.
+// Initialize the router and cache the initial page using the initial vDOM.
+// Once this code is tested and more mature, the head should be updated for region based navigation as well.
+if ( process.env.IS_GUTENBERG_PLUGIN ) {
+	if ( navigationMode === 'fullPage' ) {
+		// Cache the scripts. Has to be called before fetching the assets.
```

Comment:
Why is the caching required? Is this to guarantee execution order of the scripts when the page is loaded? What if a script is not cacheable?

Reply:
Or is this rather to ensure that there is no flash of unrendered content?

Reply:
I believe that the original idea was simply that when doing a full-page navigation to another page, we don't want to re-download the scripts that were already present on the initial page.

If a script is not cacheable (e.g. if some of the script's contents are generated dynamically) that's indeed currently a problem. 

One solution could be examine the HTTP headers and e.g. skip the ones with `Cache-Control: no-store` , `Cache-Control: no-cache` or `Expires: <Date in the past>`

Reply:
Thanks, @michalczaplinski, for chiming in. 😄

We want to revisit the whole assets system, and examining HTTP headers is something we could try.

BTW, I've created a tracking issue to continue with the development. https://github.com/WordPress/gutenberg/issues/60951

Reply:
Great! Indeed, we should take a holistic view of this. Thanks for creating the issue! 🙏